### PR TITLE
[Repositories] Add more precise types to repository generator

### PR DESCRIPTION
**Pull request by mackal**
datatypes

This adds support for unsigned and more integer types. It also avoids
using parsing functions that require casting (still needed in some
cases)

Having the data types in the Repository structs better map to the types
in the database will allow us to avoid casting when we pull data out of
them. And as a benefit, assume something is wrong if we do :)

Hopefully clean up some warnings due to casting too.

### CODE REVIEW

**Comment by Akkadius** [comment]
I'd regenerate all repos with these changes and we need to go through testing

**Comment by Akkadius** [comment]
Regenerated all base repos

**Comment by Akkadius** [comment]
Build fail

```
FAILED: common/CMakeFiles/common.dir/shared_tasks.cpp.o 
ccache /usr/bin/c++  -DBOTS -DBUILD_LOGGING -DCOMMANDS_LOGGING -DCPPHTTPLIB_OPENSSL_SUPPORT -DEMBPERL -DEMBPERL_PLUGIN -DENABLE_SECURITY -DEQEMU_USE_OPENSSL -DGLM_ENABLE_EXPERIMENTAL -DGLM_FORCE_CTOR_INIT -DGLM_FORCE_RADIANS -DHAS_UNION_SEMUN -DLUA_EQEMU -DSANITIZE_LUA_LIBS -I../common/Patches -I../common/SocketLib -I../common/StackWalker -isystem libs/zlibng -isystem /usr/include/mysql -isystem ../libs/zlibng -isystem ../submodules/glm -isystem ../submodules/cereal/include -isystem ../submodules/fmt/include -isystem ../submodules/libuv/include -isystem ../submodules/recastnavigation/DebugUtils/Include -isystem ../submodules/recastnavigation/Detour/Include -isystem ../submodules/recastnavigation/DetourCrowd/Include -isystem ../submodules/recastnavigation/DetourTileCache/Include -isystem ../submodules/recastnavigation/Recast/Include -isystem ../submodules/websocketpp -isystem /usr/include/lua5.1 -isystem ../libs/luabind -isystem /usr/lib/x86_64-linux-gnu/perl/5.28/CORE -O0 -g -DNDEBUG   -std=c++17 -MD -MT common/CMakeFiles/common.dir/shared_tasks.cpp.o -MF common/CMakeFiles/common.dir/shared_tasks.cpp.o.d -o common/CMakeFiles/common.dir/shared_tasks.cpp.o -c ../common/shared_tasks.cpp
../common/shared_tasks.cpp: In static member function 'static SharedTaskRequest SharedTask::GetRequestCharacters(Database&, uint32_t)':
../common/shared_tasks.cpp:95:73: error: no matching function for call to 'min(int&, const uint32_t&)'
   request.lowest_level  = std::min(request.lowest_level, character.level);
                                                                         ^
In file included from /usr/include/c++/8/bits/char_traits.h:39,
                 from /usr/include/c++/8/ios:40,
                 from /usr/include/c++/8/ostream:38,
                 from /usr/include/c++/8/iostream:39,
                 from ../common/eqemu_logsys.h:24,
                 from ../common/database.h:25,
                 from ../common/shared_tasks.h:4,
                 from ../common/shared_tasks.cpp:1:
/usr/include/c++/8/bits/stl_algobase.h:195:5: note: candidate: 'template<class _Tp> constexpr const _Tp& std::min(const _Tp&, const _Tp&)'
     min(const _Tp& __a, const _Tp& __b)
     ^~~
/usr/include/c++/8/bits/stl_algobase.h:195:5: note:   template argument deduction/substitution failed:
../common/shared_tasks.cpp:95:73: note:   deduced conflicting types for parameter 'const _Tp' ('int' and 'uint32_t' {aka 'unsigned int'})
   request.lowest_level  = std::min(request.lowest_level, character.level);
```

**Comment by Akkadius** [comment]
This will have a lot of ramifications for creating more explicit types - the build will need to be worked through.

Anytime we make changes to the base repository generator we should be regenning all of the **base** repositories.

File: utils/scripts/generators/repository-generator.pl [hunk 1]
```diff
@@ -491,23 +507,41 @@ sub trim {
...
+    my $mysql_column_type = $_[1];
 
     my $struct_data_type = "std::string";
-    if ($mysql_data_type =~ /tinyint/) {
-        $struct_data_type = 'int';
-    }
-    elsif ($mysql_data_type =~ /smallint/) {
-        $struct_data_type = 'int';
+    if ($mysql_column_type =~ /unsigned/) {
+        if ($mysql_data_type =~ /bitint/) {
```
Comment (hgtw) [inline utils/scripts/generators/repository-generator.pl:514]: typo for `bigint`

**Comment by hgtw** [comment]
I have no issue with either changing `task_activities.step` to signed or changing the structs to unsigned. I guess I'd lean more towards just using int but your call. Since that method is templated to (hackily) support the different zone and world structs they just need to match to avoid casting.

I guess for levels it'd be easier to just change the local request variables to uint32_t to match signs in character_data, though realistically int will hold all player levels. I don't really have an issue with either choice

**Comment by mackal** [comment]
I'm aware there is still 1 more issue to solve for this, I just haven't gotten around to it.

**Review by Akkadius — APPROVED** [review]

**Pull request merged**